src/waypoint.jsx

```diff
@@ -19,46 +19,46 @@ const Waypoint = React.createClass({
   /**
    * @return {Object}
    */
-  getDefaultProps: function() {
+  getDefaultProps() {
     return {
       threshold: 0,
-      onEnter: function() {},
-      onLeave: function() {},
+      onEnter() {},
+      onLeave() {},
     };
   },
 
-  componentDidMount: function() {
-    this.scrollableParent = this._findScrollableParent();
-    this.scrollableParent.addEventListener('scroll', this._handleScroll);
+  componentDidMount() {
+    this.scrollableAncestor = this._findScrollableAncestor();
+    this.scrollableAncestor.addEventListener('scroll', this._handleScroll);
     window.addEventListener('resize', this._handleScroll);
     this._handleScroll();
   },
 
-  componentDidUpdate: function() {
+  componentDidUpdate() {
     // The element may have moved.
     this._handleScroll();
   },
 
-  componentWillUnmount: function() {
-    if (this.scrollableParent) {
-      // At the time of unmounting, the scrollable parent might no longer exist.
-      // Guarding against this prevents the following error:
+  componentWillUnmount() {
+    if (this.scrollableAncestor) {
+      // At the time of unmounting, the scrollable ancestor might no longer
+      // exist. Guarding against this prevents the following error:
       //
       //   Cannot read property 'removeEventListener' of undefined
-      this.scrollableParent.removeEventListener('scroll', this._handleScroll);
+      this.scrollableAncestor.removeEventListener('scroll', this._handleScroll);
       window.removeEventListener('resize', this._handleScroll);
```

Comment:
Shouldn't the removal of the `resize` listener from the `window` always occur, regardless of whether or not the `scrollableAncestor` is mounted?